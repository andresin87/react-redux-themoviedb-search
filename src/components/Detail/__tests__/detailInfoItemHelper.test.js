import { getItemContent } from '../detailInfoItemHelper';

describe('DetailInfoItem helper', () => {
  describe('Array', () => {
    it(`Array is null`, () => {
      const item = {
        key: 'production_companies',
        label: 'Prod. companies',
        type: 'array',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });

    it(`Array is empty`, () => {
      const item = {
        key: 'production_companies',
        label: 'Prod. companies',
        type: 'array',
        value: [],
      };
      expect(getItemContent(item)).toMatchSnapshot();
    });

    it(`Array has empty items`, () => {
      const item = {
        key: 'production_companies',
        label: 'Prod. companies',
        type: 'array',
        value: [
          {
            id: 711,
            logo_path: '/tEiIH5QesdheJmDAqQwvtN60727.png',
            name: null,
            origin_country: 'US',
          },
          {
            id: 20555,
            logo_path: null,
            name: 'Taurus Film',
            origin_country: '',
          },
        ],
      };
      expect(getItemContent(item)).toMatchSnapshot();
    });
  });

  describe('Currency', () => {
    it(`Currency is null`, () => {
      const item = {
        key: 'budget',
        label: 'Budget',
        type: 'currency',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });
  });

  describe('Date', () => {
    it(`Date is null`, () => {
      const item = {
        key: 'release_date',
        label: 'Release date',
        type: 'date',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });

    it(`Date has wrong pattern`, () => {
      const item = {
        key: 'release_date',
        label: 'Release date',
        type: 'date',
        value: '199-10-15',
      };
      expect(getItemContent(item)).toMatchSnapshot(`Item ${item.key}`);
    });
  });

  describe('Number', () => {
    it(`Number is null`, () => {
      const item = {
        key: 'popularity',
        label: 'Popularity',
        type: 'number',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });
  });

  describe('Time', () => {
    it(`Time is null`, () => {
      const item = {
        key: 'runtime',
        label: 'Runtime',
        type: 'time',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });
  });

  describe('Text', () => {
    it(`Text is null`, () => {
      const item = {
        key: 'overview',
        label: 'Overview',
        type: 'text',
        value: null,
      };
      expect(getItemContent(item)).toEqual('');
    });
  });
});
