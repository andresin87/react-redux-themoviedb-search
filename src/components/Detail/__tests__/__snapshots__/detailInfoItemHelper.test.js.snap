// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfoItem helper Array Array has empty items 1`] = `" Taurus Film"`;

exports[`DetailInfoItem helper Array Array is empty 1`] = `""`;

exports[`DetailInfoItem helper Date Date has wrong pattern: Item release_date 1`] = `"null"`;
